src/Model/ChangeStreamIterator.php

```diff
@@ -244,9 +246,19 @@ private function extractResumeToken($document)
             return $this->extractResumeToken($document->bsonSerialize());
         }
 
-        $resumeToken = is_array($document)
-            ? ($document['_id'] ?? null)
-            : ($document->_id ?? null);
+        if ($document instanceof Document) {
+            $resumeToken = $document->has('_id')
+                ? $document->get('_id')
+                : null;
+
+            if ($resumeToken instanceof Document) {
+                $resumeToken = $resumeToken->toPHP();
```

Comment:
It seems you've considered several situations when writing this code, leading to multiple code execution paths. Are they all tested?

Reply:
@alcaeus: Did you address this comment?

Reply:
Sorry, that comment seems to have gotten lost.

Since our tests extract the `_id` field through command subscribers which always convert BSON to PHP values, I've decided to convert the resume token to PHP values in the iterator as well. This makes it easier for us to assert against the extracted resume token in tests.